java/gateway-core/src/main/java/com/avaulta/gateway/rules/Endpoint.java

```diff
@@ -55,6 +55,16 @@ public Optional<Set<String>> getAllowedMethods() {
         return Optional.ofNullable(allowedMethods);
     }
 
+    //if provided, headers provided will be pass-through to the endpoint
+    // NOTE: Using List, as Set is not being serializable in YAML
```

Comment:
This is interesting; I have to put `List` instead of `Set` -which originally I've used- because I've discovered that is not being serialized in YAML. Then, I'll adapt https://app.asana.com/0/1205303754976321/1205322459838014 in a different PR, because even the change is simple, the changes are going to be huge (all YAMLs + rules structure files)